### PR TITLE
test: setup CI and framework for unit tests

### PR DESCRIPTION
**Pull request by anujsinha3**
#92 

- [x] add frontend CI pipeline and unit test framework for flutter.
- [x] added unit tests to validate signupcodes
- [x] added build_runner step in CI pipeline for testing using mock classes

### CODE REVIEW

File: .github/workflows/ci_support_sphere_flutter.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Frontend CI
+
+on:
```
Comment (lsetiawan) [inline .github/workflows/ci_support_sphere_flutter.yml:3]: Could you make this workflow to run only for changes within `src/support_sphere`

Comment (lsetiawan) [reply .github/workflows/ci_support_sphere_flutter.yml:3]: Also, probably should run on push to main as well

File: src/support_sphere/test/unit_tests/auth_service_test.dart [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+void main() {
+
+  late AuthService authService;
+
+  setUp(() {
+    authService = AuthService();
+  });
+
+  group('AuthService', () {
```
Comment (lsetiawan) [inline src/support_sphere/test/unit_tests/auth_service_test.dart:13]: What is `group` vs `setUp`? Could you provide a little bit of description for these two? Thanks!

File: .github/workflows/ci_support_sphere_flutter.yml (hunk 1 above)
Comment (anujsinha3) [reply .github/workflows/ci_support_sphere_flutter.yml:3]: Updated

File: src/support_sphere/test/unit_tests/auth_service_test.dart (hunk 2 above)
Comment (anujsinha3) [reply src/support_sphere/test/unit_tests/auth_service_test.dart]: Updated

**Review by lsetiawan — APPROVED** [review]
LGTM

**Pull request merged**